lib/src/kvs/tx.rs

```diff
@@ -2298,4 +2298,67 @@ impl Transaction {
 		}
 		Ok(())
 	}
+
+	// set_timestamp_for_versionstamp correlates the given timestamp with the current versionstamp.
+	// This allows get_versionstamp_from_timestamp to obtain the versionstamp from the timestamp later.
+	pub(crate) async fn set_timestamp_for_versionstamp(
+		&mut self,
+		ts: u64,
+		ns: &str,
+		db: &str,
+		lock: bool,
+	) -> Result<(), Error> {
+		// This also works as an advisory lock on the ts keys so that there is
+		// on other concurrent transactions that can write to the ts_key or the keys after it.
+		let vs = self.get_timestamp(crate::key::database::vs::new(ns, db), lock).await?;
```

Comment:
In other words, every SurrealDB can safely and concurrently run this `set_timestamp_for_versionstamp` function, even before we introduce a task ownership/management/distribution system later @phughk 😸 

As we've discussed internally based on my RFC, the correctness is guaranteed using the KVS transactions, not by the task ownership or the leader election mechanism.

Reply:
Thats really cool, thanks @mumoshu ! ⏰👑😄